### ISSUE TITLE
onGetUserProfile never executes

### ISSUE THREAD

**Issue by Avrgebro** [d3xt3r2909/linkedin_login#25]
Hi, i've been trying to use this library to authenticate with Linkedin, but whenever a user authenticates himself in the webview, the expected behavior would be to close the webview and execute the code inside the onGetUserProfile callback.

However, what is actually happening is that the user gets redirected to the redirectUri page, and the webview never closes, thus never executing the onGetUserProfile or catchError

the code is basically the same as the example for the library:

```
 Navigator.push(
    context,
    MaterialPageRoute(
    builder: (BuildContext context) => LinkedInUserWidget(
        appBar: AppBar(
            title: Text('Login con LinkedIn'),
        ),
        destroySession: false,
        redirectUrl: DotEnv().env['LINKEDIN_REDIRECT_URI'],
        clientId: DotEnv().env['LINKEDIN_CLIENT_ID'],
        clientSecret: DotEnv().env['LINKEDIN_CLIENT_SECRET'],
        onGetUserProfile: (LinkedInUserModel linkedInUser) {
            print('Access token ${linkedInUser.token.accessToken}');
            print('User id: ${linkedInUser.userId}');
            Navigator.pop(context);
        },
        catchError: (LinkedInErrorObject error) {
            print('Error description: ${error.description},'
                ' Error code: ${error.statusCode.toString()}');
            Navigator.pop(context);
        },
    ),
    fullscreenDialog: true,
    ),
);
```
Please help

**Comment by d3xt3r2909** [comment]
Hi @Avrgebro  and thanks for opening a ticket. 

It's very hard for me to reproduce it with this info. I have tried to copy and paste your code into my example and it works. This means that issue can be something with redirection URL, clientId, or clientSecret. 

Is it possible for you to provide me a full example of project or we can communicate this somewhere on private if you cannot share your info? 

Cheers

**Comment by Avrgebro** [comment]
Hey @d3xt3r2909 thanks for your answer.
I tested my code with some other linkedin credentials and it worked as expected.

Apparently the page my redirecturl redirected to, had some scripts and logs that prevented the webview to properly return info to the flutter app, that's what was causing the issue. I changed the redirectUrl and it worked flawlessly.

Great library, really simple to use.
Thanks a lot!!
